blocks/library/list/index.js

```diff
@@ -72,6 +112,17 @@ registerBlock( 'core/list', {
 		return (
 			<Editable
 				tagName={ nodeName.toLowerCase() }
+				onConfig={ ( settings ) => ( {
+					...settings,
+					plugins: ( settings.plugins || [] ).concat( 'lists' ),
+					lists_indent_on_tab: false,
+				} ) }
+				onSetup={ ( editor ) => {
+					editor.on( 'nodeChange', ( nodeInfo ) => {
+						setAttributes( { internalListType: findInternalListType( nodeInfo ) } );
+					} );
+					setAttributes( { editor } );
```

Comment:
I haven't really settled on whether we should encourage attributes as a generic object store. One issue and potential reason for not leaning on it heavily is that the default behavior of serializing block comments is to encode any additional attributes outside a block's own `attributes` property into the comment itself. You can see this when changing from Visual to Text, that the block separation includes `editor` and `internalListType`:

```html
<!-- wp:core/list editor="[object Object]" internalListType="null" -->
```

The [proposed block API](https://github.com/WordPress/gutenberg/tree/master/blocks#api) includes an `encodeAttributes` function which was intended to address this concern, allowing block implementers to omit properties not desired to be included in the comment. This has not yet been implemented though. If it were, I might suggest that it be an _opt-in_ than an _opt-out_ to the comment attributes.

What do you think?

We could leave this for now and address separately. This is also why I'd considered using `state` in the `edit: class extends Component` gist for editor and internal list type.